Kernel/Arch/aarch64/RegisterState.h

```diff
@@ -11,35 +11,124 @@
 #include <AK/Platform.h>
 VALIDATE_IS_AARCH64()
 
+#include <AK/Types.h>
+#include <Kernel/Arch/CPU.h>
+
 namespace Kernel {
 
 struct RegisterState {
-    u64 x[31];     // Saved general purpose registers
+    u64 x[32];     // Saved general purpose registers
     u64 spsr_el1;  // Save Processor Status Register, EL1
     u64 elr_el1;   // Exception Link Register, EL1
     u64 tpidr_el1; // EL0 thread ID
     u64 sp_el0;    // EL0 stack pointer
 
-    FlatPtr userspace_sp() const { return 0; }
+    FlatPtr pc;
+    FlatPtr nzcv;
```

Comment:
This needs a corresponding change in https://github.com/SerenityOS/serenity/blob/master/Kernel/Arch/aarch64/vector_table.S

However, I'm not sure why there are suddenly 32 general purpose registers? I'm pretty sure there are 31 of them.

Reply:
checked in the docs (cf. [AArch64 special registers](https://developer.arm.com/documentation/den0024/a/ARMv8-Registers/AArch64-special-registers)), youʼre right here! 🙇🏻‍♂️

i was thinking of this facultative use of sp register but it seems itʼs only a part of encoding scheme, so this approach with ‘x31ʼ might be more of greater use in instruction decoder for assembly; iʼll remove it from here.